### DISCUSSION TITLE
How to save query parameters after redirect to the index page Laravel + Inertia js

### DISCUSSION THREAD

**Discussion by flight1990** [inertiajs/inertia#1003] [Help]
Hello! I have some issue. For example. I have the index page with table of users also I have some filters like sorting or searching. When I apply some filters all is good. Inertia send get request to end point, changing url query parameters for example user?search=alex
but when I click on a link to a single user's viewing page and then come back all filters are reset and url looks like /user.
How can I save my pevious url /user?search=alex???
A similar situation occurs after the user is deleted.
First I filter the data to find all the administrators url looks like /user?search=admin, after that I delete user. Then controoler redirect me back to route users.index and url looks again like /users

**Comment by druc** [comment]
Hey @flight1990 

Make sure you redirect *back*, not to the `users.index` route.

```
return redirect()->back(); 
```

**Reply by flight1990** [reply]
yes, after saving, updating or deleting it will probably solve the problem, but for example, if I go to the user's viewing page, then how do I go back?

**Reply by druc** [reply]
I think clicking the browser's back button should work just fine - unless you want to somehow maintain those filters at all times (even when navigating to any/multiple routes). Then you'd need to either save&reuse the filters from localstorage, or save & grab them from the session.
